shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -930,6 +936,16 @@ - (UIInputViewController*)inputViewController {
   return _textInputPlugin.textInputDelegate;
 }
 
+- (BOOL)respondsToSelector:(SEL)selector {
+  if (@available(iOS 17.0, *)) {
+    // See the comment on this method.
+    if (selector == @selector(insertionPointColor)) {
+      return NO;
+    }
+  }
+  return [FlutterTextInputView instancesRespondToSelector:selector];
```

Comment:
Why isn't this just calling super?

Reply:
According to the documentation `[super respondsToSelector:]` is equivalent to `[self respondsToSelector:]` https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418583-respondstoselector#discussion, so i thought there was going to be an infinite loop. Let me try it out.

Reply:
Well that worked. Changed to calling super.

Reply:
The context there is important:

> You cannot test whether an object inherits a method from its superclass by sending respondsToSelector: to the object using the super keyword. This method will still be testing the object as a whole, not just the superclass’s implementation. Therefore, sending respondsToSelector: to super is equivalent to sending it to self.

For the purposes of testing where a method's implementation lives, it's equivalent. That doesn't mean it doesn't follow actual method dispatch rules.